vms/platformvm/txs/builder/builder.go

```diff
@@ -661,3 +672,39 @@ func (b *builder) NewTransferSubnetOwnershipTx(
 	}
 	return tx, tx.SyntacticVerify(b.ctx)
 }
+
+func (b *builder) NewBaseTx(
+	amount uint64,
+	owner secp256k1fx.OutputOwners,
+	keys []*secp256k1.PrivateKey,
+	changeAddr ids.ShortID,
+) (*txs.Tx, error) {
+	ins, outs, _, signers, err := b.Spend(b.state, keys, amount, b.cfg.TxFee, changeAddr)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't generate tx inputs/outputs: %w", err)
+	}
```

Comment:
This should be handled more similarly to how `NewExportTx` works. Currently this method is willing to use locked + stakeable assets for `amount` - but those would fail to verify.